tests/test_ensemble.py

```diff
@@ -0,0 +1,242 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import itertools
+import typing as t
+
+import pytest
+
+from smartsim.entity import _mock
+from smartsim.entity._new_ensemble import Ensemble
+from smartsim.entity.strategies import ParamSet
+from smartsim.settings.launchCommand import LauncherType
+from smartsim.settings.launchSettings import LaunchSettings
+
+pytestmark = pytest.mark.group_a
```

Comment:
What is this: `pytestmark = pytest.mark.group_a`?

Reply:
It's a shortcut that `pytest` offers to turn this:
```py
@pytest.mark.group_a
def test_a():
    ...

@pytest.mark.group_a
def test_b():
    ...

@pytest.mark.group_a
def test_c()
```
into
```py
pytestmark = pytest.mark.group_a
# all tests in this module are now a part of `group_a`

def test_a(): ...
def test_b(): ...
def test_c(): ...
```

Our CI got split up into `group_a`, `group_b`, `slow_tests` and `backends` a while back in an attempt to reduce the time it takes to evaluate.

Unfortunately, it means that in order for a test to get automatically run by the CI it needs to be manually added to one of those groups (something we haven't enforced super well in the past and at least part of the reason I would be okay with removing the groups from the CI).

For now, since all of our tests are going pretty fast, I have just been adding everything into `group_a` so that the CI will continuously run the checks on each commit.